alf/algorithms/dynamics_learning_algorithm.py

```diff
@@ -74,24 +75,22 @@ def __init__(self,
 
         self._action_spec = action_spec
         self._feature_spec = feature_spec
-
-        feature_dim = flat_feature_spec[0].shape[-1]
+        self._num_replicas = num_replicas
 
         if isinstance(hidden_size, int):
             hidden_size = (hidden_size, )
 
         if dynamics_network is None:
             encoded_action_spec = TensorSpec((self._num_actions, ),
                                              dtype=torch.float32)
-            dynamics_network = EncodingNetwork(
+            dynamics_network = DynamicsNetwork(
                 name="dynamics_net",
                 input_tensor_spec=(feature_spec, encoded_action_spec),
                 preprocessing_combiner=NestConcat(),
                 fc_layer_params=hidden_size,
-                last_layer_size=feature_dim,
-                last_activation=math_ops.identity)
+                output_tensor_spec=flat_feature_spec[0])
 
-        self._dynamics_network = dynamics_network
+        self._dynamics_network = dynamics_network.make_parallel(num_replicas)
```

Comment:
How does this affect the speed for num_replicas=1?

Reply:
Now changed to not creating parallel network when num_replicas=1.